src/databricks/labs/ucx/source_code/directfs_access.py

```diff
@@ -33,105 +36,105 @@ class DirectFsAccess:
     source_id: str = UNKNOWN
     source_timestamp: datetime = datetime.fromtimestamp(0)
     source_lineage: list[LineageAtom] = field(default_factory=list)
-    job_id: int = -1
-    job_name: str = UNKNOWN
-    task_key: str = UNKNOWN
     assessment_start_timestamp: datetime = datetime.fromtimestamp(0)
     assessment_end_timestamp: datetime = datetime.fromtimestamp(0)
 
+    def from_dict(self, data: dict[str, Any]):
+        return DirectFsAccess(**data)
+
     def replace_source(
         self,
         source_id: str | None = None,
         source_lineage: list[LineageAtom] | None = None,
         source_timestamp: datetime | None = None,
     ):
-        return DirectFsAccess(
-            path=self.path,
-            is_read=self.is_read,
-            is_write=self.is_write,
-            source_id=source_id or self.source_id,
-            source_timestamp=source_timestamp or self.source_timestamp,
-            source_lineage=source_lineage or self.source_lineage,
-            job_id=self.job_id,
-            job_name=self.job_name,
-            task_key=self.task_key,
-            assessment_start_timestamp=self.assessment_start_timestamp,
-            assessment_end_timestamp=self.assessment_start_timestamp,
-        )
+        data = {
+            **asdict(self),
+            "source_id": source_id or self.source_id,
+            "source_timestamp": source_timestamp or self.source_timestamp,
+            "source_lineage": source_lineage or self.source_lineage,
+        }
+        return self.from_dict(data)
+
+    def replace_assessment_infos(
+        self, assessment_start: datetime | None = None, assessment_end: datetime | None = None
+    ):
+        data = {
+            **asdict(self),
+            "assessment_start_timestamp": assessment_start or self.assessment_start_timestamp,
+            "assessment_end_timestamp": assessment_end or self.assessment_end_timestamp,
+        }
+        return self.from_dict(data)
```

Comment:
why not dataclasses.replace(self,source_id=....)?

Reply:
because it leads to lots of duplication

Reply:
done